src/plugins/lineendings/testmod_lineendings.c

```diff
@@ -46,8 +66,25 @@ void testinvalid (const char * file)
 	KeySet * conf = ksNew (20, keyNew ("system:/valid", KEY_VALUE, "CRLF", KEY_END), KS_END);
 	PLUGIN_OPEN ("lineendings");
 	KeySet * ks = ksNew (0, KS_END);
-	succeed_if (plugin->kdbGet (plugin, ks, parentKey) == 1, "kdbget failed");
-	succeed_if (plugin->kdbSet (plugin, ks, parentKey) == (-1), "should have failed");
+
+	succeed_if (keyGetMeta (parentKey, "warnings") == NULL, "A warning on the parentKey was present before calling kdbGet.");
+
+	/* get value */
+	succeed_if (plugin->kdbGet (plugin, ks, parentKey) == ELEKTRA_PLUGIN_STATUS_NO_UPDATE,
+		    "kdbGet returned an unexpected value (ELEKTRA_PLUGIN_STATUS_NO_UPDATE was expected).");
+
+	/* check if a warning was added to the parent key */
+	succeed_if (keyGetMeta (parentKey, "warnings") != NULL,
+		    "A warning on the parentKey was not present after an invalid call to kdbGet.");
+
+
+	succeed_if (keyGetMeta (parentKey, "error") == NULL, "An error on the parentKey was present before the call the kdbSet");
+
+	/* set value */
+	succeed_if (plugin->kdbSet (plugin, ks, parentKey) == ELEKTRA_PLUGIN_STATUS_ERROR,
+		    "kdbSet returned an unexpected value (ELEKTRA_PLUGIN_STATUS_ERROR was expected).");
+
+	succeed_if (keyGetMeta (parentKey, "error") != NULL, "An error on the parentKey was not present after an invalid call to kdbSet.");
```

Comment:
Please improve tests so that it gets more clear what the tests are doing. (What exactly in the file is wrong etc.)